drivers/backup/auth.go

```diff
@@ -213,12 +228,28 @@ func getKeycloakEndPoint(admin bool) (string, error) {
 }
 
 // GetPxBackupNamespace returns namespace of px-backup deployment.
-func GetPxBackupNamespace() string {
+func GetPxBackupNamespace() (string, error) {
 	ns, present := os.LookupEnv(pxBackupNamespace)
-	if !(present) {
-		return AdminTokenSecretNamespace
+	var allServices *corev1.ServiceList
+	var err error
+	if allServices, err = k8sCore.ListServices("", metav1.ListOptions{}); err != nil {
+		if present {
+			return "", fmt.Errorf("failed to get PxBackup service [%s] in [%s] namespace. Err: %v", PxBackupServiceName, pxBackupNamespace, err)
+		}
+		return "", fmt.Errorf("failed to get list of services. Err: %v", err)
+	}
+	for _, svc := range allServices.Items {
+		if svc.Name == PxBackupServiceName {
```

Comment:
Let us reduce the complexity of this function and remove the environment variable `pxBackupNamespace`

This means irrespective of the env variable it will always look for the service and get the correct namespace.

Reply:
We now directly look for px-backup service in available namespaces.